scripts/train/run_batch.sh

```diff
@@ -13,32 +13,38 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+_env_to_tf_var_mig () {
+    echo "project_id = \"${PROJECT_ID:?}\""
+    echo "resource_prefix = \"${NAME_PREFIX:?}\""
+    echo "target_size = ${INSTANCE_COUNT:?}"
+    echo "zone = \"${ZONE:?}\""
+
+    if [ -n "${GPU_COUNT}" ] && [ -n "${ACCELERATOR_TYPE}" ]; then
+        echo "guest_accelerator = { count = ${GPU_COUNT}, type = \"${ACCELERATOR_TYPE}\" }"
+    fi
+    if [ -n "${LABELS}" ]; then echo "labels = ${LABELS}"; fi
+    if [ -n "${IMAGE_FAMILY_NAME}" ]; then
+        echo "machine_image = { family = \"${IMAGE_FAMILY_NAME}\", name = null, project = \"ml-images\" }"
+    elif [ -n "${IMAGE_NAME}" ]; then
+         echo "machine_image = { family = null, name = \"${IMAGE_NAME}\", project = \"ml-images\" }"
+    fi
+    if [ -n "${VM_TYPE}" ]; then  echo "machine_type = \"${VM_TYPE}\""; fi
+
+    if [ -n "${METADATA}" ]; then echo "metadata = ${METADATA}"; fi
+    if [ -n "${STARTUP_COMMAND}" ]; then echo "startup_script=\"${STARTUP_COMMAND}\""; fi
+}
+
 _invoke_cluster_tool () {
   echo "Invoking cluster tool"
-  echo PROJECT_ID $PROJECT_ID
-  echo NAME_PREFIX $NAME_PREFIX
-  echo ZONE $ZONE
-  echo INSTANCE_COUNT $INSTANCE_COUNT
-  echo GPU_COUNT $GPU_COUNT
-  echo VM_TYPE $VM_TYPE
-  echo ACCELERATOR_TYPE $ACCELERATOR_TYPE
-  echo IMAGE_FAMILY_NAME $IMAGE_FAMILY_NAME
-  echo IMAGE_NAME $IMAGE_NAME
-  echo DISK_SIZE_GB $DISK_SIZE_GB
-  echo DISK_TYPE $DISK_TYPE
-  echo TERRAFORM_GCS_PATH $TERRAFORM_GCS_PATH
-  echo VM_LOCALFILE_DEST_PATH $VM_LOCALFILE_DEST_PATH
-  echo METADATA $METADATA
-  echo LABELS $LABELS
-  echo STARTUP_COMMAND $STARTUP_COMMAND
-  echo ORCHESTRATOR_TYPE $ORCHESTRATOR_TYPE
-  echo GCS_MOUNT_LIST $GCS_MOUNT_LIST
-  echo NFS_FILESHARE_LIST $NFS_FILESHARE_LIST
-  echo SHOW_PROXY_URL $SHOW_PROXY_URL
-  echo MINIMIZE_TERRAFORM_LOGGING $MINIMIZE_TERRAFORM_LOGGING
-  echo NETWORK_CONFIG $NETWORK_CONFIG
-  echo ACTION $ACTION
-  /usr/entrypoint.sh
+  echo '```tfvars'
+  cat "${TFVARS_FILE}"
+  echo '```'
+  /root/aiinfra/scripts/entrypoint.sh \
+      ${MINIMIZE_TERRAFORM_LOGGING:+ --quiet} \
+      ${TERRAFORM_GCS_PATH:+ --backend-bucket ${TERRAFORM_GCS_PATH}} \
+      ${ACTION,,} \
+      mig \
```

Comment:
Is this supposed to be hardcoded to `mig`?

Reply:
The only clusters available are `["gke", "mig", "mig-with-container", "slurm"]`; the other script seemed like it was specifically for `gke`, and variables required for `mig-with-container` and `slurm` are not present in this script, so I kinda just assumed it to be specifically for `mig`. I might be missing something though, should I make it variable?